external-crates/move/crates/move-disassembler/src/disassembler.rs

```diff
@@ -300,7 +404,9 @@ impl<'a> Disassembler<'a> {
     // defined in the module in question.
     fn disassemble_struct_def(
         &self,
-        buffer: &mut impl Write,
+        buffer: &mut (impl Write + ByteLength),
+        bcode_map_gen: bool,
+        bcode_map: &mut SourceMap,
```

Comment:
Sorry to be so nitty, but I only mention it as I think it will clean up the logic...

What about passing `bcode_map: &mut Option<SourceMap>` here. Then below, instead of things like

```rust
if bcode_map_gen {
   bcode_map.add_top_level_struct_mapping(...)
}
```

This could be rewritten as

```rust
bcode_map.map(|m| m.add_top_level_struct_mapping(..));
```

or even as 
```rust
if let Some(m) = bcode_map {
   m.add_top_level_struct_mapping(...);
}
```

Reply:
Good idea - will also let me shave off a parameter in a few places